src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -401,7 +415,18 @@ void VideoDecoder::addVideoStreamDecoder(
   activeStreamIndices_.insert(streamNumber);
   updateMetadataWithCodecContext(streamInfo.streamIndex, codecContext);
   streamInfo.options = options;
-  initializeFilterGraphForStream(streamNumber, options);
+  int width = options.width.value_or(codecContext->width);
+  // There is a bug in sws_scale where it doesn't handle non-multiple of 32
+  // widths.
+  // https://stackoverflow.com/questions/74351955/turn-off-sw-scale-conversion-to-planar-yuv-32-byte-alignment-requirements
+  // In that case we are forced to use a filtergraph to do the color conversion.
+  auto colorConversionLibrary = options.colorConversionLibrary.value_or(
+      ColorConversionLibrary::FILTERGRAPH);
+  bool useFilterGraph =
+      (colorConversionLibrary != ColorConversionLibrary::SWSCALE);
```

Comment:
Based on the comment above, I would expect to see something like:
```
if (width % 32 != 0) {
  useFilterGraph = true;
}
```
I think I understand why: for benchmarking, we want absolute to use sws_scale even if the results are wrong. That is, we want the benchmark to control which color conversion library we use, not the C++ library.

I think that's okay, so long as it only happens through `_add_video_stream()`. I would prefer that we defensively add the above check in `add_video_stream()` now, rather than waiting until we implement `ColorConversionLibrary::AUTO`. This sort of bug is the kind that is easy to miss, and I'd like us to be extra defensive. The comment above, with the reference to StackOverflow, would then make the most sense there.

Reply:
I removed the comment as it was just a remnant from before.

I hardcoded "filtergraph" in the public API.

Reply:
@ahmadsharif1, where is it hardcoded? I was expecting to find it in `core.add_video_stream()`, but I didn't see it there.